src/hex.ts

```diff
@@ -55,6 +65,50 @@ export function assertIsStrictHexString(value: unknown): asserts value is Hex {
   );
 }
 
+/**
+ * Validate that the passed prefixed hex string is a valid hex address, or a
+ * valid mixed-case checksum address.
+ *
+ * @param possibleAddress - Input parameter to check against.
+ * @returns Whether or not the input is a valid hex address.
+ */
+export function isValidHexAddress(possibleAddress: Hex) {
+  return (
+    is(possibleAddress, HexAddressStruct) ||
+    isValidChecksumAddress(possibleAddress)
+  );
+}
+
+/**
+ * Validate that the passed hex string is a valid ERC-55 mixed-case
+ * checksum address.
+ *
+ * @param possibleChecksum - The hex address to check.
+ * @returns True if the address is a checksum address.
+ */
+export function isValidChecksumAddress(possibleChecksum: Hex) {
+  if (!is(possibleChecksum, HexChecksumAddressStruct)) {
+    return false;
+  }
+
+  const unPrefixed = remove0x(possibleChecksum);
+  const unPrefixedHash = remove0x(
+    bytesToHex(keccak256(unPrefixed.toLowerCase())),
+  );
+
+  for (let i = 0; i < unPrefixedHash.length; i++) {
+    const value = parseInt(unPrefixedHash[i] as string, 16);
```

Comment:
Nit: we can avoid this cast by looping over the characters in the string rather than using a for loop. `unPrefixedHash.forEach` lets us get the index as well.

Reply:
Even looping with a `for..of` or a `.forEach` on `unPrefixed` doesn't eliminate the need for this cast, as we are using the index to take a char from a different string than the one we loop over.

This commented line is confusing because I erroneously looped over the wrong string.